common/operationmodes/operationmodes.go

```diff
@@ -9,38 +12,43 @@ const (
 	OperationModeLiteObserver       = "lite-observer"
 )
 
-// CheckOperationModes will check the compatibility of the provided operation modes and return an error if any
-func CheckOperationModes(modes []string) error {
+// ParseOperationModes will check and parse the operation modes
+func ParseOperationModes(operationModeList string) ([]string, error) {
+	if len(operationModeList) == 0 {
+		return []string{}, nil
+	}
+
+	modes := strings.Split(operationModeList, ",")
 	if len(modes) == 0 {
-		return nil
+		return []string{}, nil
```

Comment:
unreachable code. You can test to see I'm right :D

Reply:
right. removed